ipyleaflet/leaflet.py

```diff
@@ -816,6 +822,21 @@ class Heatmap(RasterLayer):
     blur = Float(15.0).tag(sync=True, o=True)
     gradient = Dict({0.4: 'blue', 0.6: 'cyan', 0.7: 'lime', 0.8: 'yellow', 1.0: 'red'}).tag(sync=True, o=True)
 
+    def __init__(self, **kwargs):
+        super(Heatmap, self).__init__(**kwargs)
+        self.data = self._get_data()
+
+    @observe('gradient')
+    def _updata_data(self, change):
+        self.data = self._get_data()
+
+    def _get_data(self):
+        values = list(self.gradient.keys())
+        colors = list(self.gradient.values())
+        self.vmin = values[0]
+        self.vmax = values[-1]
```

Comment:
I think we need to sort the gradient by value first.

Reply:
Sorry. I am not sure to understand.

Reply:
`gradient` is a dict of "value:color" entries. I don't think we can expect the first entry to have the minimum value and the last entry to have the maximum value.

Reply:
I meant that we need to sort the `colormap_labels`. What happens to `vmin` and `vmax` if `gradient = {1.0: 'red', 0.6: 'cyan', 0.7: 'lime', 0.8: 'yellow', 0.4: 'blue'}`?

Reply:
> I meant that we need to sort the `colormap_labels`. What happens to `vmin` and `vmax` if `gradient = {1.0: 'red', 0.6: 'cyan', 0.7: 'lime', 0.8: 'yellow', 0.4: 'blue'}`?

You're right. Branca colormap requires that the `colormaps_labels` are sorted. This point has been solved in commit [373f5d9](https://github.com/jupyter-widgets/ipyleaflet/pull/1036/commits/373f5d9b8816892d50080cd9fe009a9f43e89f1b)
The example notebook has also been modified for a case where the gradient is initially not sorted.